subprojects/execution/src/main/java/org/gradle/internal/execution/impl/OutputUtil.java

```diff
@@ -35,66 +36,106 @@
 import org.gradle.internal.snapshot.SnapshotVisitResult;
 
 import javax.annotation.Nullable;
-import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 import java.util.function.BiPredicate;
 
+import static com.google.common.collect.ImmutableSortedMap.copyOfSorted;
+import static com.google.common.collect.Maps.transformEntries;
 import static org.gradle.internal.snapshot.MerkleDirectorySnapshotBuilder.EmptyDirectoryHandlingStrategy.EXCLUDE_EMPTY_DIRS;
 import static org.gradle.internal.snapshot.MerkleDirectorySnapshotBuilder.EmptyDirectoryHandlingStrategy.INCLUDE_EMPTY_DIRS;
+import static org.gradle.internal.snapshot.SnapshotUtil.index;
 
-/**
- * Filters out fingerprints that are not considered outputs. Entries that are considered outputs are:
- * <ul>
- * <li>an entry that did not exist before the execution, but exists after the execution</li>
- * <li>an entry that did exist before the execution, and has been changed during the execution</li>
- * <li>an entry that did wasn't changed during the execution, but was already considered an output during the previous execution</li>
- * </ul>
- */
-public class OutputFilterUtil {
+public class OutputUtil {
 
-    public static ImmutableList<FileSystemSnapshot> filterOutputSnapshotBeforeExecution(FileCollectionFingerprint afterLastExecutionFingerprint, FileSystemSnapshot beforeExecutionOutputSnapshot) {
-        Map<String, FileSystemLocationFingerprint> fingerprints = afterLastExecutionFingerprint.getFingerprints();
+    /**
+     * Filters out snapshots that are not considered outputs. Entries that are considered outputs are:
+     * <ul>
+     * <li>an entry that did not exist before the execution, but exists after the execution</li>
+     * <li>an entry that did exist before the execution, and has been changed during the execution</li>
```

Comment:
I think this Javadoc doesn't make much sense on the method, since there is no change during the execution. What are we actually filtering here?

Reply:
Let's update this together.